src/main/java/org/jetbrains/research/refactorinsight/folding/RefactoringFolder.java

```diff
@@ -0,0 +1,127 @@
+package org.jetbrains.research.refactorinsight.folding;
+
+import com.intellij.application.options.colors.highlighting.RendererWrapper;
+import com.intellij.codeInsight.daemon.impl.HintRenderer;
+import com.intellij.diff.FrameDiffTool;
+import com.intellij.diff.requests.DiffRequest;
+import com.intellij.diff.requests.SimpleDiffRequest;
+import com.intellij.diff.tools.util.side.OnesideTextDiffViewer;
+import com.intellij.diff.tools.util.side.ThreesideTextDiffViewer;
+import com.intellij.diff.tools.util.side.TwosideTextDiffViewer;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.FoldRegion;
+import com.intellij.openapi.editor.ex.EditorEx;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiFile;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.research.refactorinsight.adapters.RefactoringType;
+import org.jetbrains.research.refactorinsight.data.RefactoringInfo;
+import org.jetbrains.research.refactorinsight.folding.handlers.FoldingHandler;
+import org.jetbrains.research.refactorinsight.folding.handlers.MoveOperationHandler;
+import org.jetbrains.research.refactorinsight.services.MiningService;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+public class RefactoringFolder {
+  static Map<RefactoringType, FoldingHandler> foldingHandlers;
+
+  static {
+    foldingHandlers = new EnumMap<>(RefactoringType.class);
+    FoldingHandler moveOperationHandler = new MoveOperationHandler();
+    foldingHandlers.put(RefactoringType.MOVE_OPERATION, moveOperationHandler);
+    foldingHandlers.put(RefactoringType.PULL_UP_OPERATION, moveOperationHandler);
+    foldingHandlers.put(RefactoringType.PUSH_DOWN_OPERATION, moveOperationHandler);
+  }
+
+  // Suppresses default constructor, ensuring non-instantiability.
+  private RefactoringFolder() {
+  }
+
+  /**
+   * TODO: Javadoc.
+   */
+  public static void foldRefactorings(@NotNull FrameDiffTool.DiffViewer viewer, @NotNull DiffRequest request) {
+    if (request instanceof SimpleDiffRequest) {
+      SimpleDiffRequest diffRequest = (SimpleDiffRequest) request;
+      String commitId = diffRequest.getContentTitles().get(1);
+      if (viewer instanceof OnesideTextDiffViewer) {
+        RefactoringFolder.foldRefactorings((OnesideTextDiffViewer) viewer, commitId);
+      } else if (viewer instanceof TwosideTextDiffViewer) {
+        RefactoringFolder.foldRefactorings((TwosideTextDiffViewer) viewer, commitId);
+      } else if (viewer instanceof ThreesideTextDiffViewer) {
+        RefactoringFolder.foldRefactorings((ThreesideTextDiffViewer) viewer, commitId);
+      }
+    }
+  }
+
+  private static void foldRefactorings(@NotNull OnesideTextDiffViewer viewer, String commitId) {
```

Comment:
Please, investigate which of them we really need to fold refactorings that relate to methods. I guess we need only `TwosideTextDiffViewer`, but I'm not sure